src/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -142,6 +142,104 @@ public StageManagementResource(string stageServiceUri, HttpSource httpSource)
             return result;
         }
 
+        public async Task<StageDetailedView> Get(string id, Common.ILogger log)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException(StagingResources.StageIdShouldNotBeEmpty, nameof(id));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            var result = await _httpSource.ProcessResponseAsync(
+                () =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stageServiceUri, $"{StagePath}/{id}"));
+
+                    return request;
+                },
+                async response =>
+                {
+                    await EnsureSuccessStatusCode(response);
+
+                    string responseBody = await response.Content.ReadAsStringAsync();
+                    return JsonConvert.DeserializeObject<StageDetailedView>(responseBody);
+                },
+                log,
+                CancellationToken.None);
+
+            return result;
+        }
+
+        public async Task Commit(string id, string apiKey, Common.ILogger log)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException(StagingResources.StageIdShouldNotBeEmpty, nameof(id));
+            }
+
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentException(StagingResources.ApiKeyShouldNotBeEmpty, nameof(apiKey));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            await _httpSource.ProcessResponseAsync(
+                () =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Post, new Uri(_stageServiceUri, $"{StagePath}/{id}"));
+                    request.Headers.Add(ApiKeyHeader, apiKey);
+
+                    return request;
+                },
+                async response =>
+                {
+                    await EnsureSuccessStatusCode(response);
+                    return true;
+                },
+                log,
+                CancellationToken.None);
+        }
+
+        public async Task<StageCommitProgressView> GetCommitProgress(string id, Common.ILogger log)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException(StagingResources.StageIdShouldNotBeEmpty, nameof(id));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            var result = await _httpSource.ProcessResponseAsync(

```

Comment:
this looks very similar to the Get method above. you may want to create a helper method that just takes the url path constructed here. or at least move the verify json response helper to a static method that both can use